Task 3/Task3.2/Task3.2/DYNAMICARRAY.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+namespace Task3._2
+{
+    class DynamicArray<T> : IEnumerable<T>
+    {
+        internal T[] _array;
+        int _capacity;
+        int _lenght = -1;
+        public T this[int index]
+        {
+            get
+            {
+                if (index > Lenght)
+                    throw new ArgumentOutOfRangeException(nameof(index), "OutOfRangeException");
+                return _array[index];
+            }
+
+            set => _array[index] = value;
+        }
+
+        public int Capacity
+        {
+            get => _capacity;
+            set
+            {
+                _capacity = value;
+                var tempArray = new T[_capacity];
+                for (int i = 0; i < _array.Length; i++)
+                {
+                    tempArray[i] = _array[i];
+                }
+                _array = tempArray;
+            }
+        }
+        public int Lenght 
+        {
+            get => _lenght;
+            private set
+            {
+                _lenght = value;
+                if (Capacity <= _lenght)
+                {
+                    int tempCount = Capacity;
+                    do
+                    {
+                        tempCount *= 2;
+                    } while (tempCount < _lenght);
+                    Capacity = tempCount;
+                }
+            }
+        }
+
+        public DynamicArray()
+        {           
+            _array = new T[8];            
+            Capacity = 8;
```

Comment:
You can get capacity as array length - less chances to forget something and make a mistake